README.md

```diff
@@ -0,0 +1,104 @@
+# FMS HF Tuning
+
+This repo aims to create basic tuning scripts with support for specific models, assumes the use of Hugging Face `SFTTrainer` and PyTorch FSDP only. Our approach to tuning is:
+1. Models are loaded from Hugging Face `transformers` or the `foundation-model-stack` -- models are either optimized to use `Flash Attention v2` directly or through `SDPA`
+2. Hugging Face `SFTTrainer` for the training loop
+3. `FSDP` as the backend for training
+
+## Installation
+
+```
+pip install -r requirements.txt
+pip install -U datasets
+pip install -e .
+```
+
+## Data format
+The data format expectation is a single column text. The trainer is configured to expect a response template as a string. For example, if one wants to prepare the `alpaca` format data to feed into this trainer, it is quite easy and can be done with the following code.
+
+```python
+PROMPT_DICT = {
+    "prompt_input": (
+        "Below is an instruction that describes a task, paired with an input that provides further context. "
+        "Write a response that appropriately completes the request.\n\n"
+        "### Instruction:\n{instruction}\n\n### Input:\n{input}\n\n### Response:"
+    ),
+    "prompt_no_input": (
+        "Below is an instruction that describes a task. "
+        "Write a response that appropriately completes the request.\n\n"
+        "### Instruction:\n{instruction}\n\n### Response:"
+    ),
+}
+
+def format_alpaca_fn(example):
+    prompt_input, prompt_no_input = PROMPT_DICT['prompt_input'], PROMPT_DICT['prompt_no_input']
+    output = prompt_input.format_map(example) if example.get("input", "") != "" else prompt_no_input.format_map(example)
+    output = f"{output} {example['output']}"
+    return {"output": output}
```

Comment:
I'm not seeing how we handle `IGNORE_INDEX` / `-100` for the parts of the prompt that should be ignored in the backward pass

Reply:
This is being done in the HF codebase based on a `response_template`. Does the assumption of `IGNORE_INDEX` being the default `-100` pose issues? I do see that their `DataCollator` is exposing this in API with default of `-100`. We can fix this as a parameter based on the model:

Relevant loc: https://github.com/huggingface/trl/blob/main/trl/trainer/utils.py#L178